plugins/coverbrowser.koplugin/bookinfomanager.lua

```diff
@@ -228,6 +228,35 @@ function BookInfoManager:loadSettings()
     for i, key in ipairs(keys) do
         self.settings[key] = values[i]
     end
+
+    -- translate old append_series_to_authors and append_series_to_title to new series_mode
+    if not self.settings["series_mode"] then -- initialize from old settings
```

Comment:
Good point. I considered just dropping backwards compatibility, but that's something that tends to annoy me, so I didn't want to do it to others :)

Reply:
(Not resolved).
I insist: you don't need any SQL in there. Just use `saveSetting()`.

There's no real backward compatibility needed, except for users that downgrade from time to time (I do when checking for regressions - but I woudln't mind much losing that series setting :)
I wouldn't even mind if we were not migrating it :) Users would notice and would then go see that menu - and see your new option.

But if you do want migration, and want to keep the old settings, and avoid recursion when dealing with them, just use:
```lua
if not self.settings["zzz_migration_series_setting_migrated"] then
    self:saveSetting("zzz_migration_series_setting_migrated", true)
        -- should stop re-doing the following when saveSetting() re-loadSettings()
    local series_mode
    if self.settings["append_series_to_authors"] then
        series_mode = "append_series_to_authors"
    end -- no elseif, the user may have both
    if self.settings["append_series_to_title"] then
        series_mode = "append_series_to_title"
    end
    self:saveSetting("series_mode", series_mode)
end
```

Reply:
I don't want to keep the old settings, migrating is sufficient.
The reason I didn't call saveSetting() is because it calls loadSettings(), which leads to a 3-deep recursion if both flags are set. I do get your point about not replicating SQL code, though, so I'll adapt the code.

Reply:
I'm having second thoughts, and cannot decide between keeping obsolete append_* settings lying around and using series_mode="none" instead of series_mode=nil.
Which variant do you prefer?

Reply:
Personally, I would just prefer if there was no migration code :) (I don't mind if we don't migrate them - as we would then not carry that migration code for years).
Otherwise, I have no preference (they are stored in `bookinfo_cache.sqlite3` that no one looks at - I would mind just a tad more if they would be stored in G_reader_settings, but not much :)
After that, `series_mode=nil` makes just more sense if you're not ever interested in series.